Renaming throw type and exception parameter err msg to incompatible

Ao-senXiong opened the pull request:
I will do the two renaming first as they are the most related to our changes in #436.

wmdietl reviewed (changes requested):
Did you look through the manual whether this needs changing somewhere?
This definitely needs a changelog entry.
Can we support both suppression keys for a transition period?

wmdietl commented:
> > Can we support both suppression keys for a transition period?
> 
> I have added back to old keys for this transition period.

You left the old keys in `messages.properties`, where nobody uses them. You can remove them there.

Can we allow existing `@SuppressWarnings` that use the old keys to continue to work? Otherwise, we're breaking everyone that currently suppresses these warnings and force them to change the keys.

Ao-senXiong commented:
> > > Can we support both suppression keys for a transition period?
> > 
> > 
> > I have added back to old keys for this transition period.
> 
> You left the old keys in `messages.properties`, where nobody uses them. You can remove them there.
> 
> Can we allow existing `@SuppressWarnings` that use the old keys to continue to work? Otherwise, we're breaking everyone that currently suppresses these warnings and force them to change the keys.


Addressed this comment in [7c26ec1](https://github.com/eisop/checker-framework/pull/618/commits/7c26ec1c1a2dc1a6096aaa1f9f022366602a141d)
